### ISSUE TITLE
pre-trained googleNet Error using vl_taccummex

### ISSUE THREAD

**Issue by ncalik** [vlfeat/matconvnet#1034]
Hi everyone,
Firstly I would like to thank you very much for your contributions to the forum.
I have started CNN with matconvnet and done many new ConvNet architecture as sequential and dagnn with your favour...

For my research, I want to use googleNet pretrained model and use above code for it...

netStruct = load('imagenet-googlenet-dag.mat') ;

net                 = dagnn.DagNN.loadobj(netStruct) ;
layerName     = 'cls3_fc';
layerIndex      = net.getLayerIndex(layerName);
paramNames = [net.layers(layerIndex).params];

paramIdx = net.getParamIndex(paramNames); 
 net.params(paramIdx(1)).value = single(randn(1,1,1024,2));
 net.params(paramIdx(2)).value = single(randn(2,1));
 
 net.layers(153)=[]; %remove the softmax layer

% add following loss layers
net.addLayer('loss', dagnn.Loss('loss', 'log'), ...
{'cls3_fc','label'}, 'objective') ;
net.addLayer('top1err', dagnn.Loss('loss', 'classerror'), ...
{'cls3_fc','label'}, 'top1err') ;

When I print layer ind in dagnn.DagNN/eval, I observed that all layer can be processed. But updating time, I got an error like : 

![image](https://user-images.githubusercontent.com/17113517/29290038-634ef592-8147-11e7-9099-5aea0583caab.png)

p is params index of net, [0 0] is value of a and [1 1 512 128] is value of b. 
I couldnt find any solution. Can anyone help for this problem? My question is also same with #451
 
And also i have a little question too.
How can we use adam solver?
I tried params.solver = @adam but not work..

Thanks again..

**Comment by maxiaoyuelun** [comment]
@ncalik do you solve this problem?

**Comment by maxiaoyuelun** [comment]
@ncalik googlenet have three losses, but I've no idea about how to write the losses, just like #1003

**Comment by firatozgenel** [comment]
@ncalik @maxiaoyuelun 
Hi, I had similar problems with GoogleNet and Error using vl_taccumex. I shared the code that I used in #451 .
Hope this helps. 
Best

**Comment by dengyuhk** [comment]
@ncalik  do you solve this problem yet？
